packages/prisma/store/PRISMA_MIGRATION_TRACKING.md

```diff
@@ -0,0 +1,258 @@
+# Prisma Tenant-Aware Migration Tracking
+
+This document tracks the migration of all files that need to be converted to use tenant-aware Prisma clients. **All files must be migrated before merging the PR.**
+
+## Migration Summary
+
+| Category | Total Files | Migrated | Remaining |
+|----------|-------------|----------|-----------|
+| API Routes (App Router) | 58 | 0 | 58 |
+| API Routes (Pages Router) | 121 | 0 | 121 |
+| NestJS API (v2) | 32 | 0 | 32 |
+| tRPC Procedures | 145 | 145 | 0 |
+| Repository Classes | 35 | 0 | 35 |
+| Utility Functions | 168 | 0 | 168 |
+| Test Files | 112 | 0 | 112 |
+| Cron Jobs | 35 | 0 | 35 |
+| **Total** | **706** | **145** | **561** |
+
+## Migration Patterns
+
+### 1. API Routes (App Router)
+
+**Migration Pattern**: Use `withPrismaRoute` HOC
+
+**Before:**
+```typescript
+import prisma from "@calcom/prisma";
+
+export async function GET(req: Request) {
+  const users = await prisma.user.findMany();
+  return Response.json({ users });
+}
+```
+
+**After:**
+```typescript
+import { withPrismaRoute } from "@calcom/prisma/store/withPrismaRoute";
+import type { PrismaClient } from "@prisma/client";
+
+async function handler(req: Request, prisma: PrismaClient) {
+  const users = await prisma.user.findMany();
+  return Response.json({ users });
+}
+
+export const GET = withPrismaRoute(handler);
+```
+
+### 2. API Routes (Pages Router)
+
+**Migration Pattern**: Use `withPrismaApiHandler` HOC
+
+**Before:**
+```typescript
+import prisma from "@calcom/prisma";
+
+export default async function handler(req, res) {
+  const users = await prisma.user.findMany();
+  res.status(200).json({ users });
+}
+```
+
+**After:**
+```typescript
+import { withPrismaApiHandler } from "@calcom/prisma/store/withPrismaApiHandler";
+import type { PrismaClient } from "@prisma/client";
+
+async function handler(req, res, prisma: PrismaClient) {
+  const users = await prisma.user.findMany();
+  res.status(200).json({ users });
+}
+
+export default withPrismaApiHandler(handler);
```

Comment:
DevinAI, I know we already have a prismaStore for each tenant. But is it possible to also store the "current" tenant prisma? That way we can call it anywhere where it's wrapped and always get the current tenant aware prisma client. The host indentification should happen once in the entrypoint and then store the result so we avoid prop drilling and we can inject the client deep down the code. Here's some pseudo-code on how the API should look like.

```suggestion
async function handler(req, res) {
  const prisma = getTenantAwarePrisma();
  const users = await prisma.user.findMany();
  res.status(200).json({ users });
}

export default withPrismaApiHandler(handler);
```